src/java/org/apache/cassandra/service/snapshot/TableSnapshot.java

```diff
@@ -113,54 +139,171 @@ public long computeSizeOnDiskBytes()
 
     public long computeTrueSizeBytes()
     {
-        return snapshotDirs.stream().mapToLong(trueDiskSizeComputer::apply).sum();
+        DirectorySizeCalculator visitor = new SnapshotTrueSizeCalculator();
+
+        for (File snapshotDir : snapshotDirs)
+        {
+            try
+            {
+                Files.walkFileTree(snapshotDir.toPath(), visitor);
+            }
+            catch (IOException e)
+            {
+                logger.error("Could not calculate the size of {}. {}", snapshotDir, e);
```

Comment:
Should not this be replaced by `error(String msg, Throwable t)`? I do not think that this version of the method is the one we want. Sorry if I am wrong.

Reply:
Good catch, fixed it.